core/src/main/java/com/graphhopper/GHRequest.java

```diff
@@ -88,9 +94,87 @@ public GHRequest addPoint( GHPoint point )
                     + "more than two places via addPlace method.");
 
         points.add(point);
+        preferredDirections.add(preferredDirection);
+        return this;
+    }
+
+    public GHRequest addPoint( GHPoint point)
+    {
+        addPoint(point, Double.NaN);
+        return this;
+    }
+
+    /**
+     * initialize directions list with non-preferred directions
+     */
+    private void initDirectionsList( int size )
+    {
+        preferredDirections = new ArrayList<Double>(Collections.nCopies(size, Double.NaN));
+    }
+
+    /**
+     * Sets preferred directions for starting (start point) and ending (via and end points)
+     * Directions are north based azimuth (clockwise) in (0,2pi) or NaN for equal preference
+     */
+    public GHRequest setPreferredDirections(List<Double> directions)
+    {
+      if (points.size() != directions.size())
+      {
+          throw new IllegalArgumentException("Size of directions (" + directions.size() + 
+                  ") must match size of points (" + points.size() + ")"); 
+      }
+      for (Double direction : directions)
+      {
+          validateAzimuthValue(direction);
+      }
+      this.preferredDirections = directions;
+      return this;
+    }
+
+    /**
+     * Sets preferred direction for a specific point of the request
+     * @param direction north based azimuth (clockwise) in (0,2pi) or NaN for equal preference
+     * @param position point for which direction is set. Starts with 0 for the start point  
+     */
+    public GHRequest setPreferredDirection(Double direction, int position)
```

Comment:
it is not necessary.
I just thought it is more convenient for using the low level API.  For example specifying only preferred start direction (without caring how many other points there are)